src/include/optimizer/statistics/table_stats_collector.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <vector>
+
+#include "optimizer/stats/column_stats_collector.h"
+#include "catalog/schema.h"
+
+namespace terrier::optimizer {
+//===--------------------------------------------------------------------===//
+// TableStatsCollector
+//===--------------------------------------------------------------------===//
+class TableStatsCollector {
+ public:
+  TableStatsCollector(storage::DataTable* table);
+
+  ~TableStatsCollector();
+
+  void CollectColumnStats();
+
+  inline size_t GetActiveTupleCount() { return active_tuple_count_; }
+
+  inline size_t GetColumnCount() { return column_count_; }
+
+  ColumnStatsCollector* GetColumnStats(oid_t column_id);
```

Comment:
The name of the method is `GetColumnStats` but it returns a `ColumnStatsCollector` instead of a `ColumnStats` object.